File: DSA/SparshJain/Day_3_Q1.cpp
#include <iostream>
#include <string>
#include <unordered_set>

std::string checkPangram(const std::string& text) {
    std::unordered_set<char> chars;

    for (char c : text) {
        if (isalpha(c)) {
            chars.insert(tolower(c));
        }
    }

    if (chars.size() == 26) {
        return "complete";
    } else {
        return "incomplete";
    }
}

int main() {
    std::string sentence = "The quick brown fox jumps over the lazy dog";
    std::cout << checkPangram(sentence) << std::endl;

    return 0;
}

File: DSA/Shruti-Pandey/Day_3_q1.java
public class PangramChecker {
    public static String checkPangram(String sentence) {
        sentence = sentence.toLowerCase();
        boolean[] letterPresent = new boolean[26];
        for (char c : sentence.toCharArray()) {
            if (c >= 'a' && c <= 'z') {
                letterPresent[c - 'a'] = true;
            }
        }
        for (boolean present : letterPresent) {
            if (!present) {
                return "incomplete";
            }
        }
        
        return "complete";
    }
    
    public static void main(String[] args) {
        String sentence = "The quick brown fox jumps over the lazy dog";
        System.out.println(checkPangram(sentence));
    }
}
